File: app/src/main/java/com/example/features/main/ViewPagerAdapter.java
package com.example.features.main;

import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.List;

class ViewPagerAdapter extends FragmentStatePagerAdapter {
	private final List<Fragment> mFragmentList;

	ViewPagerAdapter(@NonNull FragmentManager fm, @NonNull List<Fragment> fragments) {
		super(fm);
		mFragmentList = fragments;
	}

	@Override
	public Fragment getItem(int position) {
		return mFragmentList.get(position);
	}

	@Override
	public int getCount() {
		return mFragmentList.size();
	}
}

File: app/src/main/kotlin/com/example/features/main/ViewPagerAdapter.kt
package com.example.features.main

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

internal class ViewPagerAdapter(fm: FragmentManager, private val mFragmentList: List<Fragment>) : FragmentStatePagerAdapter(fm) {
  override fun getItem(position: Int): Fragment {
    return mFragmentList[position]
  }

  override fun getCount(): Int {
    return mFragmentList.size
  }
}
